### PR TITLE
D20 detector scan fixes

### PR DESCRIPTION
**Pull request by gvardany**
**Description of work.**

Fixes few issues with D20 concerning the merge of multiple detector scans with different scan steps.

**To test:**
Use the newly added test files. Remove your `~/.mantid/instrument`.
- [ ] should run with no errors
```
scan = PowderILLDetectorScan(Run='167339:167340')
```
- [ ] should run with no errors
```
PowderILLEfficiency(CalibrationRun='167339',
                                   OutputWorkspace='calib',
                                   InterpolateOverlappingAngles=True,
                                   OutputResponseWorkspace='response')
```
- [ ] In the output of the snippet below, the sample log named `2theta.position` should have 1099 entries.
```
ws = LoadAndMerge(Filename='167339+167340')
```

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/CurveFitting/src/Functions/CubicSpline.cpp [hunk 1]
```diff
@@ -179,9 +179,9 @@ void CubicSpline::calculateSpline(double *out, const double *xValues,
     }
   }
 
-  // warn user than some values wern't calculated
+  // warn user that some values wern't calculated
```
Comment (darominski) [inline Framework/CurveFitting/src/Functions/CubicSpline.cpp:182]: typo in the original comment: wern't -> weren't

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PowderILLEfficiency.py [hunk 2]
```diff
@@ -344,7 +356,7 @@ def _configure_sequential(self, raw_ws):
             Configures the calibration with SequentialSummedReference1D method (D20)
             @param : the name of the raw detector scan (merged) workspace
         """
-        self._scan_points = mtd[raw_ws].getRun().getLogData('ScanSteps').value
+        self._scan_points = int(mtd[raw_ws].getRun().getLogData('ScanSteps').value)
```
Comment (darominski) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PowderILLEfficiency.py:359]: Is this cast necessary? It seems that the value type is always an integer.

File: docs/source/release/v6.0.0/diffraction.rst [hunk 3]
```diff
@@ -29,6 +29,7 @@ New features
 Improvements
 ############
 - :ref:`WANDPowderReduction <algm-WANDPowderReduction>` now accepts a sequence of input workspaces, combining them to reduce to a single spectrum.
+- :ref:`PowderILLDetectorScan <algm-PowderILLDetectorScan>` is fixed when treating more than one scans merged.
```
Comment (darominski) [inline docs/source/release/v6.0.0/diffraction.rst:32]: Can you rephrase to make clear it is about a) Multiple scans merged into one file, b) 'fixed' means corrected? For example 'fixed issue when treating merging multiple scans'.

**Review by darominski — COMMENTED** [review]
Changes and tests are working as described, raised just a few tiny proposed changes.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PowderILLEfficiency.py (hunk 2 above)
Comment (gvardany) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PowderILLEfficiency.py:359]: Yes, this is a calculated log, so might be float depending on the scan settings.

**Review by darominski — DISMISSED** [review]
All good now.

File: Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/PowderILLDetectorScanTest.py [hunk 4]
```diff
@@ -47,5 +46,25 @@ def test_default_options_d2b(self):
         self.assertTrue(item.getRun())
         self.assertTrue(item.getRun().hasProperty('run_number'))
 
+    def test_default_options_d20(self):
```
Comment (DavidFair) [inline Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/PowderILLDetectorScanTest.py:49]: So I'm happy with all the changes here however hesitant on this test falling afoul of our [unit test good practice](https://developer.mantidproject.org/UnitTestGoodPractice.html#other-more-general-points) and our unit tests slowing down by lots of little regressions like this (akin to "death by a thousand paper cuts")

Prior to this change the PowderILLDetectorScanTest takes 6s on [isis-ndw1297](https://builds.mantidproject.org/job/pull_requests-ubuntu/39762/consoleFull) (which isn't great) but adding this takes an additional 2 seconds on the [same machine](https://builds.mantidproject.org/job/pull_requests-ubuntu/39675/consoleFull).

Could we either par down 171414 to be a representative subset of the data (say using 10% of the original spectra) or move it to a system test?

**Review by DavidFair — CHANGES_REQUESTED** [review]

File: Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/PowderILLDetectorScanTest.py (hunk 4 above)
Comment (gvardany) [reply Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/PowderILLDetectorScanTest.py:49]: It depends, it happens that if you reduce/mock the data the test passes, but it fails with the real data. I'm not saying this is the case here, but it is always a tradeoff between skinning your data or keeping it real. In general, the strategy was to have a unit test that covers the default options, this is what is added, and then have more complete scenarios in the system test. There are many other unit test suits that run over 10s, so for me up 10s have always been acceptable. We can of course discuss later if we want to introduce stricter criteria project-wide, but I don't think it should block this PR.

Comment (DavidFair) [reply Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/PowderILLDetectorScanTest.py:49]: I agree that it's not fair to block this PR if you have already tried reducing / mocking the data, typically when I bring this up people haven't realised the impact of putting a full dataset in.

On a local developer machine it's less apparent, but you tend to find when a release is looming that one CI job that randomly failed and now will take another 2 hours to run unit tests tends to stir up annoyance. But equally that's something for developers as a whole to raise if it becomes a problem.

**Review by DavidFair — APPROVED** [review]
